cpp/include/cudf/utilities/type_dispatcher.hpp

```diff
@@ -82,6 +82,39 @@ struct id_to_type_impl {
 template <cudf::type_id Id>
 using id_to_type = typename id_to_type_impl<Id>::type;
 
+/**
+ * @brief "Returns" the corresponding type that is stored on the device when using `cudf::column`
+ *
+ * Use this "type function" with the `using` type alias:
+ * @code
+ * using Type = device_storage_type_t<Element>;
+ * @endcode
+ *
+ * @tparam T The literal type that is stored on the host
+ */
+// clang-format off
+template <typename T>
+using device_storage_type_t =
+  std::conditional_t<std::is_same<numeric::decimal32, T>::value, int32_t,
+  std::conditional_t<std::is_same<numeric::decimal64, T>::value, int64_t, T>>;
+// clang-format on
+
+/**
+ * @brief Checks if `fixed_point`-like types have template type `T` matching the column's
+ * stored type id
+ *
+ * @tparam T The type that is stored on the device
+ * @param id The `data_type::id` of the column
+ * @return true If T matches the stored column type id
+ * @return false If T does not match the stored column type id
+ */
+template <typename T>
+bool type_id_matches_device_storage_type(type_id const& id)
+{
+  return (id == type_id::DECIMAL32 && std::is_same<T, int32_t>::value) ||
+         (id == type_id::DECIMAL64 && std::is_same<T, int64_t>::value) || id == type_to_id<T>();
```

Comment:
This looks like it should use `device_storage_type_t`.

Reply:
wow :frowning_face:  so sad I didn't see this. I blame this on snow-blindness (temporarily blinded for looking at something too long) at this point :upside_down_face: 

fixed.